### PR TITLE
[Messenger] Fix redis last error not cleared between calls

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Not clearing it gives misleading errors coming from previous calls which makes debugging hard.
@alexander-schranz FYI

### CODE REVIEW

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Else it looks good for me 👍.

File: src/Symfony/Component/Messenger/Tests/Transport/RedisExt/ConnectionTest.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
 use Symfony\Component\Messenger\Transport\RedisExt\Connection;
 
 /**
- * @requires extension redis
+ * @requires extension redis >= 4.3.0
```
[the remark this answers is not in the document]
Comment (chalasr) [reply src/Symfony/Component/Messenger/Tests/Transport/RedisExt/ConnectionTest.php]: 4.2.0 is buggy on debian, we added a check that requires 4.3+. See #31791

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

File: src/Symfony/Component/Messenger/Tests/Transport/RedisExt/ConnectionTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply src/Symfony/Component/Messenger/Tests/Transport/RedisExt/ConnectionTest.php]: @chalasr thank you! This is good to know.

**Pull request merged**